nautobot/ipam/models.py

```diff
@@ -378,15 +381,40 @@ def clean(self):
                     {"location": f'Prefixes may not associate to locations of type "{self.location.location_type}".'}
                 )
 
+    def delete(self, **kwargs):
+        force_delete = kwargs.pop("force_delete", False)
+
+        try:
+            super().delete(**kwargs)
+        except models.ProtectedError as err:
+            # Update protected objects to use the new parent and delete the old parent (self).
+            if force_delete:
+                protected_pks = (po.pk for po in err.protected_objects)
+                protected_objects = Prefix.objects.filter(pk__in=protected_pks)
+                protected_objects.update(parent=self.parent)
+                super().delete(**kwargs)
```

Comment:
`delete()` actually has a return signature, though I think we're not respecting it in a number of other places already.

```suggestion
            return super().delete(**kwargs)
        except models.ProtectedError as err:
            # Update protected objects to use the new parent and delete the old parent (self).
            if force_delete:
                protected_pks = (po.pk for po in err.protected_objects)
                protected_objects = Prefix.objects.filter(pk__in=protected_pks)
                protected_objects.update(parent=self.parent)
                return super().delete(**kwargs)
```

Reply:
Yes, thanks. Good catch.